ts/src/main/scala/ch/epfl/ts/example/FullMarketSimulation.scala

```diff
@@ -0,0 +1,148 @@
+package ch.epfl.ts.example
+
+import ch.epfl.ts.component.{ComponentRef, StopSignal, ComponentBuilder}
+import com.typesafe.config.ConfigFactory
+import akka.actor.Props
+import ch.epfl.ts.brokers.StandardBroker
+import scala.language.postfixOps
+import scala.concurrent.duration._
+import ch.epfl.ts.data._
+import ch.epfl.ts.traders.{MadTrader, MovingAverageTrader, SimpleTraderWithBroker}
+import ch.epfl.ts.data.Register
+import ch.epfl.ts.engine._
+import ch.epfl.ts.data.MarketBidOrder
+import ch.epfl.ts.data.Register
+import ch.epfl.ts.engine.FundWallet
+import ch.epfl.ts.data.MarketAskOrder
+import ch.epfl.ts.component.fetch.{HistDataCSVFetcher, PullFetchComponent, TrueFxFetcher, MarketNames}
+import scala.reflect.ClassTag
+import ch.epfl.ts.data.Currency._
+import ch.epfl.ts.data.MarketBidOrder
+import ch.epfl.ts.data.Register
+import ch.epfl.ts.engine.FundWallet
+import ch.epfl.ts.data.Quote
+import ch.epfl.ts.data.Currency
+import ch.epfl.ts.data.MarketAskOrder
+import ch.epfl.ts.component.utils.BackLoop
+import ch.epfl.ts.component.persist.DummyPersistor
+import java.util.Timer
+import ch.epfl.ts.engine.rules.{SimulationMarketRulesWrapper, FxMarketRulesWrapper}
+
+/**
+ * Market simulation with first reading historical data and then running simulation on its own.
+ * Created by sygi on 08.05.15.
+ */
+object FullMarketSimulation {
+  var producers = Map[Class[_], List[ComponentRef]]()
+  var consuments = Map[Class[_], List[ComponentRef]]()
```

Comment:
It's actually a popular [terminology](http://en.wikipedia.org/wiki/Producer%E2%80%93consumer_problem)